### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by blackwhitebear8** [wills106/homeassistant-solax-modbus#214]
**Describe the bug**
I m getting: unrecognized Solis inverter type - serial number : unknown and Solis: reading serial number from address 0x80ec failed; other address may succeed

I m using a DSD TECH SH-U10 rs485 to usb adaptor it has a silicon labs cp2102 chip in it. its correctly wired up to our solis 10KW Hybird inverter

**Mandatory details**
1. Integration Version: 0.7.0.5
2. HA Version: Home Assistant 2022.11.4
3. Inverter Brand: solis
4. Plugin Used: solis
5. First 6 digits of Serial: 110CA2
6. Connection Method: Serial (DSD TECH SH-U10 rs485 )

**Detailed Error Log**
Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:84
Integration: SolaX Inverter Modbus ([documentation](https://github.com/wills106/homsassistant-solax-modbus), [issues](https://github.com/wills106/homsassistant-solax-modbus/issues))
First occurred: 14:50:20 (1 occurrences)
Last logged: 14:50:20

Solis: reading serial number from address 0x80ec failed; other address may succeed

Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:86
Integration: SolaX Inverter Modbus ([documentation](https://github.com/wills106/homsassistant-solax-modbus), [issues](https://github.com/wills106/homsassistant-solax-modbus/issues))
First occurred: 14:50:20 (1 occurrences)
Last logged: 14:50:20

Solis: cannot find serial number, even not for other Inverter

Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:97
Integration: SolaX Inverter Modbus ([documentation](https://github.com/wills106/homsassistant-solax-modbus), [issues](https://github.com/wills106/homsassistant-solax-modbus/issues))
First occurred: 14:50:20 (1 occurrences)
Last logged: 14:50:20

unrecognized Solis inverter type - serial number : unknown

If there are no log's showing for SolaX try the following depending on your language settings, you can find the full logs under:

Settings → System → Logs > at bottom of page press “LOAD FULL LOGS”

Now the full logs are loaded. If you scroll down, you will see them.
Once the full logs are shown, you can either use the search function in your browser to search for “solax” related entries or use the search entry field on top of the page.

Search for

solax

2022-11-25 15:00:04.261 WARNING (MainThread) [custom_components.solax_modbus.plugin_solis] Solis: reading serial number from address 0x80ec failed; other address may succeed
2022-11-25 15:00:04.264 ERROR (MainThread) [custom_components.solax_modbus.plugin_solis] Solis: cannot find serial number, even not for other Inverter
2022-11-25 15:00:04.265 ERROR (MainThread) [custom_components.solax_modbus.plugin_solis] unrecognized Solis inverter type - serial number : unknown

**Additional context**

**Comment by wills106** [comment]
Are you using the default baud rate of 9600?

Have you previously been able to talk to this Inverter either using the Modbus built into HA or a program like Gmodbus?

**Comment by blackwhitebear8** [comment]
I m using the default 9600 yes and no i have not tried other programs yet

**Comment by wills106** [comment]
Are you sure you have the data pins the correct way round?
With Modbus A goes to A and B goes to B. They don't need swapping over unlike a Null Modem Cable.

**Comment by blackwhitebear8** [comment]
a goes to a and b goes to b

**Comment by blackwhitebear8** [comment]
i used this picture for the pinout on the inverter itself https://community-assets.home-assistant.io/original/4X/7/5/0/750b95a33f06f15259bee9bd843ef19fc27e56b6.jpeg

Pin 3 goes to a+ and pin 4 to a-

**Comment by wills106** [comment]
@kai-seipp are you able to confirm if this Diagram is correct as I believe you have the same Inverter.
Did you have to enable anything of the Inverter as well? As there doesn't appear to be any communications if it is connected correctly?

**Comment by kai-seipp** [comment]
Hey Guys, 

The pins are correct 3 / 4 are Modbus A and B. So that is correct. What took me a while to figure out is that I had to use the "COM" Port (green) and not the MODBUS Port as one could expect... ;) 

And of cause you have to check the Address given within the inverters Menu so it matches the Modbus ID you specify within this script on setup. 

You don't need to use ground or if (to shield the wire) you don't connect it to your inverter but just to your Modbus communication module. 

Hope this helps!

**Comment by blackwhitebear8** [comment]
"The pins are correct 3 / 4 are Modbus A and B. So that is correct. What took me a while to figure out is that I had to use the "COM" Port (green) and not the MODBUS Port as one could expect... ;)" what do you mean with com port green etc? i can see the device in side linux as /dev/ttyUSB0? and on the inverter we dont have any settings for modbus or com

**Comment by blackwhitebear8** [comment]
also can this come becouse of the cable length between the inverter and the serial usb? (around 10m)

**Comment by wills106** [comment]
RS485 can go upto 1200m if Terminated correctly.

I'm not sure at what exact distance you need to ground the cable and possibly introduce a terminating resistor at.

**Comment by blackwhitebear8** [comment]
i dont have it grounded atm is that required? and if so how would i do that

**Comment by blackwhitebear8** [comment]
this is what it looks on the pc/servers side btwhttps://cdn.discordapp.com/attachments/805950455359602709/1045735535987142806/IMG_20221125_171832.jpg

**Comment by kai-seipp** [comment]
![image](https://user-images.githubusercontent.com/22188947/204043146-e260ecb1-b5da-4535-9b68-8c7414765d29.jpeg)

This image shows tie two ports on my Solis inverter. 

To set modbus address I have to go to "settings" and than to "Address". The number I define there is the modbus ID.

**Comment by blackwhitebear8** [comment]
i put the cables in the COM/CAM port does it need to be in the rs one somehow?

**Comment by kai-seipp** [comment]
![image](https://user-images.githubusercontent.com/22188947/204043381-1e445d91-2945-4406-af96-13906e9507f7.jpeg)

If you scroll in you can see the pin IDs inside the socket.

**Comment by blackwhitebear8** [comment]
looks correctly wired up to me then

**Comment by kai-seipp** [comment]
> https://cdn.discordapp.com/attachments/805950455359602709/1045735535987142806/IMG_20221125_171832.jpg

Do you have an image where we can see the labels on the ports?

**Comment by kai-seipp** [comment]
> i dont have it grounded atm is that required? and if so how would i do that

I don't think you have to ground it for 10m. But this depends on the interference on the way. But at a private home it should be ok. If you want to try: you would just have to find the ground connector on your usb dongle. And connect one of the extra wires to it. That's it. It would not be connected anywhere on the inverters side.

**Comment by blackwhitebear8** [comment]
i have re attached all the 4 wires and pins so the usb gets also 5v and ground from the inverter and lets see tomorrow what it does when the sun is backup :p becouse ye rn the inverter is offline

**Comment by wills106** [comment]
It looks like @blackwhitebear8 is using Unshielded CATx cable?
I'm not sure if grounding one of the other twisted pairs would offer much help? I would assume you would have to ground the outer shielding for it to make a difference?

Are you able to use a shorter cable just for test purposes? Could you use a laptop next to your Inverter?

I don't it's necessary to connect the 5v between devices.

Like @kai-seipp mentions are you able to show a picture of how you are connected at the Inverter?

**Comment by blackwhitebear8** [comment]
https://cdn.discordapp.com/attachments/1026751450874904598/1045782707982041170/IMG_20221125_201936.jpg
https://cdn.discordapp.com/attachments/1026751450874904598/1045782708204359800/IMG_20221125_202058.jpg
connections ^^ and its a shielded cat6 cable the shielding is cut off where the plastic stops.

**Comment by kai-seipp** [comment]
I would be careful using pins 1 and 2. I don't have it to hand but if I remember correctly one is feeding 5V and the other 24V for the Solis data stick. I did find a breakout sheet somewhere on the Solis help portal. But this is just out of my head. But anyway: there really is no need to use anything else but 3 and 4. I only used these and used totally unshielded single wires you would usually use to connect a door bell. But I only needed to bridge about 1m.

**Comment by blackwhitebear8** [comment]
i had found this diagram and it said 5v and ground https://community-assets.home-assistant.io/original/4X/7/5/0/750b95a33f06f15259bee9bd843ef19fc27e56b6.jpeg

**Comment by kai-seipp** [comment]
> i had found this diagram and it said 5v and ground https://community-assets.home-assistant.io/original/4X/7/5/0/750b95a33f06f15259bee9bd843ef19fc27e56b6.jpeg
> 
> 

Checked it with the Solis portal. Seams to be correct! 

https://ginlongsolis.freshdesk.com/support/solutions/articles/36000012141-rs-485-wiring-with-circular-connector

**Comment by kai-seipp** [comment]
Solis: 

A typical wiring scheme calls for an IN and an OUT, that means that each run will have a positive, a negative, a common wire, and a ground. The common wire isn't needed here, but the + and - are. Think of A as positive and B as negative. Each in / out has both a positive (A) or a negative (B).

**Comment by kai-seipp** [comment]
Did you check the address set within your inverter? Does it match the one set here at the integration?

**Comment by blackwhitebear8** [comment]
where can i check this? i have never seen any options for serial or modbus connections in it?

**Comment by kai-seipp** [comment]
Follow this here. https://user-images.githubusercontent.com/22188947/204049730-61f2cafc-ab85-484f-ac9d-2de318f1dbba.MOV

**Comment by blackwhitebear8** [comment]
will try this tomorrow morning ty. Also do you have abatterij or do you live somewhere where its still day now?

**Comment by kai-seipp** [comment]
I have a battery. This is why one of my two inverter is still running. 

The other one is dead now as well.

**Comment by blackwhitebear8** [comment]
we have a battery aswell but its LV and ours only accepts HV.... this wasnt clearly documented on it or its papers it came with..

**Comment by kai-seipp** [comment]
(I am a neighbor from Germany.) So also no sun here. 

I really? I didn't know Solis only accepts HW. I have a pylontech force H2.

**Comment by blackwhitebear8** [comment]
maybe this is not the right place for this but do you maybe have discord? that I can contact you on?

**Comment by wills106** [comment]
Unfortunately it gets confusing what is classed as HV and LV from a battery perspective.

I have seen the 48v Pylontech batteries rebranded as both Low Voltage and High Voltage.
As they get connected in parallel I would class them more as LV.

Unlike the SolaX which are about 105 - 110v each and get wired in series. I have 3 batteries connected to my Gen3 SolaX which is around 315v at the moment.

To confuse matters with Solis you seem to have two side by side Inverters:

RHI-nK-48ES-5G
RHI-nK-HVES-5G

(n = kW rating)

**Comment by kai-seipp** [comment]
Solis is the worse when it comes to documentation! Everything I know about their deceives I don't know from them... 

Discord: a good idea, maybe we can setup a channel for this integration. I have it on my work Mac installed. Not using it much though as we use slack for work.

**Comment by blackwhitebear8** [comment]
after setting the address it started doing more but now i m getting this:

Logger: homeassistant.util.async_
Source: util/async_.py:180
First occurred: 10:04:00 (560 occurrences)
Last logged: 10:08:39

Detected blocking call to sleep inside the event loop. This is causing stability issues. Please report issue to the custom integration author for solax_modbus doing blocking calls at custom_components/solax_modbus/__init__.py, line 277: return self._client.read_input_registers(address, count, **kwargs)

**Comment by wills106** [comment]
The blocking calls is a known issue with using RS485 - USB https://github.com/wills106/homeassistant-solax-modbus#known-issues

It's nothing to worry about. Are you getting any entities returning values now?

**Comment by blackwhitebear8** [comment]
yes i m getting data

**Comment by wills106** [comment]
You could possibly configure the logger to ignore the blocking calls?
https://www.home-assistant.io/integrations/logger/

**Comment by blackwhitebear8** [comment]
is there a way to get thise live?
<img width="295" alt="Screenshot 2022-11-26 101316" src="https://user-images.githubusercontent.com/45701225/204081398-02ae4c90-2b54-4cfd-9269-e16b37950f37.png">
i put these in here but i want to confirm these are the right ones
<img width="863" alt="Screenshot 2022-11-26 101405" src="https://user-images.githubusercontent.com/45701225/204081425-9496354d-dd96-4f8c-9162-20f163e5ccf1.png">

**Comment by wills106** [comment]
There is an example here how to have it live with an additional card. https://github.com/wills106/homeassistant-solax-modbus/wiki/Energy-Dashboard's

Top left should be Solis Grid Import Today.

**Comment by blackwhitebear8** [comment]
the live card says the house is using as much as there is being generated? nothing imported or exported

**Comment by wills106** [comment]
The Power Distribution card? or the screen shot you posted just?

**Comment by blackwhitebear8** [comment]
The Power Distribution card
<img width="375" alt="Screenshot 2022-11-26 102223" src="https://user-images.githubusercontent.com/45701225/204081676-5b940a0f-0190-4a0a-8202-c84ea83a5283.png">

**Comment by wills106** [comment]
Can you share what entities you used for it?

**Comment by blackwhitebear8** [comment]
<img width="743" alt="Screenshot 2022-11-26 102346" src="https://user-images.githubusercontent.com/45701225/204081724-d5d181a6-a34c-439a-b9a6-fd73cf3d1409.png">

**Comment by wills106** [comment]
solis_measured_power & solis_battery_power_charge don't exist.
We need to work out what the equivalent is for the Solis.

**Comment by blackwhitebear8** [comment]
also i have found out why the usage is as high as generation the 3 phase meter seems to be sending out 0 for everything to the inverter

**Comment by wills106** [comment]
For the battery try "Battery Power"

**Comment by blackwhitebear8** [comment]
we have no working battery on it yet so that will stay 0/null xd

**Comment by wills106** [comment]
If you select the correct entity it might show 0? Not sure. battery_power_charge is a SolaX name, just need to drop the _charge.

**Comment by wills106** [comment]
For the Grid try "Meter Active Power" that should be the 3 Phases combined.

**Comment by blackwhitebear8** [comment]
ye i had found that just now but it shows 0 just like on the inverter is receiving 0 from the meter for some reason

**Comment by wills106** [comment]
What about the one without meter?
"Active Power"

**Comment by wills106** [comment]
> the 3 phase meter seems to be sending out 0 for everything to the inverter

Is this a Modbus Power meter such as a Eastron or similar? Was it supplied with the Solar Installation?

**Comment by blackwhitebear8** [comment]
i have corrected the meter settings inside solis and now its sending out the correct usage

**Comment by wills106** [comment]
If some of your reading still seem out, there might be further work required to correct them. #210

**Comment by kai-seipp** [comment]
> i have corrected the meter settings inside solis and now its sending out the correct usage

I have had the same issue: My installing company did configure an Eastron Meter, but delivered a Acrel Meter... That did not work at all...

**Comment by blackwhitebear8** [comment]
anyone know how to configure a card like this with these meters? https://github.com/ulic75/power-flow-card

**Comment by kai-seipp** [comment]
for your information (from an other thread): 

I don't trust my inverters import / exports readings at all.
I use a shelly 3EM to monitor these values. So I have not yet really checked them.

But yes they do need scaling from what I can see.
Here is what I could find:

Grid Import Yesterday Shelly: 7,66 kWh
Grid Import Yesterday Solis South Inverter: 84 kWh
Grid Import Yesterday Solis North Inverter: 60 kWh

Grid Export Yesterday Shelly: 3,09 kWh
Grid Export Yesterday Solis South Inverter: 12 kWh
Grid Export Yesterday Solis North Inverter: 12 kWh

**Comment by wills106** [comment]
For the Power Flow Card have a look at https://github.com/wills106/homeassistant-solax-modbus/discussions/31 It's geared towards Solax, but should be a starting point.

**Comment by kai-seipp** [comment]
This is how I configured my Energy dashboard: 
https://dsh.re/61a868

For the Grid Information I use my Shelly entities these are total values, not daily values. 
So for your inverters readings here you would need to use the entities: 
- sensor.solis_nord_grid_export_total
- sensor.solis_nord_grid_import_total

making "solis_nord" matching your configuration. 
I fond using the total numbers here to be rubust. But I have also heard of people using the "today" values.

**Comment by blackwhitebear8** [comment]
when usin total its a bit weird becouse theres no battery so how can things move there 
<img width="353" alt="Screenshot 2022-11-26 111321" src="https://user-images.githubusercontent.com/45701225/204083574-aab8c5b4-91c6-405b-9f42-820d7e88f5c4.png">

**Comment by kai-seipp** [comment]
What is the benefit of using [this]( https://github.com/ulic75/power-flow-card) custom card instead of the official HA card?

**Comment by blackwhitebear8** [comment]
this ones live and the HA one not?

**Comment by wills106** [comment]
The HA isn't live! It's misleading, all the values are the previous hour.

**Comment by wills106** [comment]
Also you import / export should look more realistic either later in the day / tomorrow. Unless you have imported 58kW so far today?

The battery might just show there anyway regardless if you have one or not? I have only ever had a battery connected since the Dashboard was introduced, so I don't know how it would look if you didn't have a battery.

**Comment by kai-seipp** [comment]
ok, true. 
For me that had worked so far. As I also created my own dash for my current values. 

https://dsh.re/137cd
So I might look into that custom card as well for that dash.

**Comment by blackwhitebear8** [comment]
cool

**Comment by wills106** [comment]
I could never be bothered to setup the PowerFlowCard so I just stuck to the PowerDistribution one.

Doesn't look as colourful, but does the job.

![image](https://user-images.githubusercontent.com/18155231/204083900-eef49f9f-db90-4f83-b713-9517d0ea78cc.png)

**Comment by wills106** [comment]
Last time I looked into that PowerFlow card you can to split all the bi-directional sensors into input / output. Where the Power Distribution card did that Internally.

Things might have changed since though?

**Comment by blackwhitebear8** [comment]
@kai-seipp how did you creare these graphs?
<img width="781" alt="Screenshot 2022-11-26 114015" src="https://user-images.githubusercontent.com/45701225/204084616-719b6d3c-fa64-4630-9b28-090b0edbd341.png">

**Comment by wills106** [comment]
It might be worth either adding to the discussion here #211 
Or creating a new discussion? Otherwise when this issue get closed down it will be hard to share the information.

Just thinking it will be easier for others to share the knowledge then.

**Comment by blackwhitebear8** [comment]
when editing the solax intergation like polling frequenfy etc it creates new sensors instead of using the old one......

**Comment by wills106** [comment]
When we started to develop 0.7.0beta it was changed so you couldn't have the Integration twice with the same name.
But this also prevents you from updating the configuration. (We need to relook into fixing that)

For the time being you are best removing the Integration from the Integration page, restarting HA and adding it back again with your new polling rate.

This way you can rename it the same as the initial setup. If you do it this was any custom dashboards etc will still function correctly as the naming is the same.

**Comment by kai-seipp** [comment]
> @kai-seipp how did you creare these graphs?
> 
> <img width="781" alt="Screenshot 2022-11-26 114015" src="https://user-images.githubusercontent.com/45701225/204084616-719b6d3c-fa64-4630-9b28-090b0edbd341.png">
> 
> 

These are coming from HA (Energy cards)
You can use these on your dashboard by adding a custom card.

```

type: energy-usage-graph

```

You can find the documentation here: https://www.home-assistant.io/dashboards/energy/

**Comment by blackwhitebear8** [comment]
the powerflow dashboard shows grid power as 300.000W instead of like 300W like all the others do?

**Comment by wills106** [comment]
Is that only when displayed in the powerflowcard or on the Integration page?

The powerflowcard might be expecting kW and not wasts?

**Comment by kai-seipp** [comment]
I gave Power Card a try. This is the configuration that works for me. 
https://dsh.re/59b4b6

`type: custom:power-flow-card
title: Aktueller Energie Fluss
entities:
  battery: sensor.solis_nord_battery_power
  battery_charge: sensor.solis_nord_battery_soc
  grid: sensor.power_total
  solar: sensor.current_pv_power_total
  gas: sensor.recording_ractualpower
w_decimals: 1
kw_decimals: 2
watt_threshold: 2200
inverted_entities: battery`

**Comment by blackwhitebear8** [comment]
i tried your configuration but it still shows some thingsin kwh and high values? i seem to be missing the power_total sensor

**Comment by kai-seipp** [comment]
Yes, power total is a template sensor that I have created for a Shelly device. You would have to use one of your inverter entities.

**Comment by wills106** [comment]
I am going to close down as the Inverter is now responding to the Inverter following setting the correct address.

Any further issues, please raise a new issue.
